### ISSUE TITLE
client.Group.ListAssignedApplicationsForGroup returns an application properly, but gives error for nonexistent index

### ISSUE THREAD

**Issue by pputman12** [okta/okta-sdk-golang#292]
### Describe the bug?

<img width="712" alt="image" src="https://user-images.githubusercontent.com/840282/164146703-1afa7f6e-40c2-48b1-99a1-ef585dc7afbf.png">
When trying to call ListAssignedApplicationsForGroup, I get an error:
<img width="1024" alt="image" src="https://user-images.githubusercontent.com/840282/164146740-4980178c-9213-4fc4-b5b4-02c954426997.png">
I'm writing a test that creates an okta group and app, with a rule that should assign them, so I'm testing if it exists, and it does when printing it out:

<img width="948" alt="image" src="https://user-images.githubusercontent.com/840282/164146839-26cb13ac-67ac-45fe-a9e4-6c1570756374.png">

I've isolated the problem down to ListAssignedApplicationsForGroup but not sure why its complaining about index out of range.

### What is expected to happen?

its expected to just run the test successfully, since the application exists and is assigned to the group

### What is the actual behavior?

It does pull down the application, and I can see it searches for it and finds it, but I get an error saying index out of range

### Reproduction Steps?

Try to call this method with these parameters:

ctx, client, err := oktaTest.NewConnection(
        oktaCon.testOrg,
        oktaCon.testUrl,
        oktaCon.testApiToken,
    )
    if err != nil {
        fmt.Printf("Could not connect to okta %v", err)
    }

    oktaGroupApps, _, err := client.Group.ListAssignedApplicationsForGroup(ctx, groupId, nil)

oktaGroupApps will have the proper application, but it will still error out at the end. It panics and recovers somehow?

### Golang Version

1.18 and 1.18.1

### SDK Version

latest

### OS version

Darwin Patricks-MacBook-Pro.local 21.4.0 Darwin Kernel Version 21.4.0: Fri Mar 18 00:46:32 PDT 2022; root:xnu-8020.101.4~15/RELEASE_ARM64_T6000 arm64

**Comment by monde** [comment]
@pputman12 thanks for giving me a head start into debugging this. I will look into it.

**Comment by monde** [comment]
Just checking @pputman12, did you close this ticket inadvertently? I'm going to write some code to see if I can repo this in my dev env.

**Comment by pputman12** [comment]
I didn't close it inadvertently it was intentionally, I was mistaken about
the problem.

**Comment by monde** [comment]
ok, no problem! I have new integration test to add to the next release 😄

**Comment by pputman12** [comment]
I was mistaken because I made a new integration test for something I'm
doing in okta, and it caused this error.  But the problem wasn't with this
module, somehow when I run this new test, it triggers an error with an
empty slice returning from a similar method in a different test.  If I run
each test separately, I can't reproduce it, but can reproduce it every time
if I run each test together, I'll write up something on it, but my initial
report was confusing me as it came from a similar method in another test
but only happened after I made the new test.
